### ISSUE TITLE
yapf crashes on code containing a raw f-string

### ISSUE THREAD

**Issue by kylebarron** [google/yapf#537]
Yapf crashes on code that contains a raw f-string. Consider the code:
```py
import re

regex = []
for x in ['hello', 'world']:
    regex.append(rf'^({x})$')
regex = '|'.join(regex)

re.search(regex, 'world')
```

This is valid Python 3.6 code to search for the word `hello` or `world`, but yapf is unable to parse it. This looks to be a different issue than https://github.com/google/yapf/issues/342, which concerns only f-strings, and seemed to be resolved in 3.6.2. I'm using 3.6.4:

```
> python --version                                        
Python 3.6.4 :: Anaconda custom (64-bit)
```

Traceback:
```
> yapf temp.py 
Traceback (most recent call last):
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/yapflib/pytree_utils.py", line 103, in ParseCodeToTree
    tree = parser_driver.parse_string(code, debug=False)
  File "/home/kyle/local/anaconda3/lib/python3.6/lib2to3/pgen2/driver.py", line 106, in parse_string
    return self.parse_tokens(tokens, debug)
  File "/home/kyle/local/anaconda3/lib/python3.6/lib2to3/pgen2/driver.py", line 71, in parse_tokens
    if p.addtoken(type, value, (prefix, start)):
  File "/home/kyle/local/anaconda3/lib/python3.6/lib2to3/pgen2/parse.py", line 159, in addtoken
    raise ParseError("bad input", type, value, context)
lib2to3.pgen2.parse.ParseError: bad input: type=3, value="'^({x})$'", context=('', (5, 19))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/yapflib/yapf_api.py", line 128, in FormatCode
    tree = pytree_utils.ParseCodeToTree(unformatted_source)
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/yapflib/pytree_utils.py", line 109, in ParseCodeToTree
    tree = parser_driver.parse_string(code, debug=False)
  File "/home/kyle/local/anaconda3/lib/python3.6/lib2to3/pgen2/driver.py", line 106, in parse_string
    return self.parse_tokens(tokens, debug)
  File "/home/kyle/local/anaconda3/lib/python3.6/lib2to3/pgen2/driver.py", line 71, in parse_tokens
    if p.addtoken(type, value, (prefix, start)):
  File "/home/kyle/local/anaconda3/lib/python3.6/lib2to3/pgen2/parse.py", line 159, in addtoken
    raise ParseError("bad input", type, value, context)
lib2to3.pgen2.parse.ParseError: bad input: type=3, value="'^({x})$'", context=('', (5, 19))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kyle/local/anaconda3/bin/yapf", line 11, in <module>
    load_entry_point('yapf==0.20.2', 'console_scripts', 'yapf')()
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/__init__.py", line 312, in run_main
    sys.exit(main(sys.argv))
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/__init__.py", line 200, in main
    verbose=args.verbose)
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/__init__.py", line 250, in FormatFiles
    in_place, print_diff, verify, verbose)
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/__init__.py", line 275, in _FormatFile
    logger=logging.warning)
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/yapflib/yapf_api.py", line 90, in FormatFile
    verify=verify)
  File "/home/kyle/local/anaconda3/lib/python3.6/site-packages/yapf/yapflib/yapf_api.py", line 130, in FormatCode
    raise parse.ParseError(filename + ': ' + e.message)
AttributeError: 'ParseError' object has no attribute 'message'
```

**Comment by gwelymernans** [comment]
It looks like `lib2to3` allows for f-strings but not for raw f-strings...

**Comment by NightMachinary** [comment]
Any ways to just make it skip the offending line?

**Comment by tvykruta** [comment]
Same problem. I posted a repro case in  https://github.com/google/yapf/issues/451.

**Comment by bwendling** [comment]
Closed with 7c408b9d7750292760ed255f744211d1ef535668.
